src/main/scala/util/ECC.scala

```diff
@@ -157,9 +157,9 @@ class SECDEDCode extends Code
   def width(k: Int) = sec.width(k)+1
   def encode(x: UInt, poison: Bool = Bool(false)) = {
     // toggling two bits ensures the error is uncorrectable
-    val toggle_lo = poison.asUInt // a non-data bit in SEC
-    val toggle_hi = toggle_lo << sec.width(x.getWidth) // the parity bit
-    par.encode(sec.encode(x)) ^ toggle_lo ^ toggle_hi
+    val toggle_lo = Cat(poison.asUInt, poison.asUInt)
```

Comment:
Mind adding a comment to explain the rationale?

Reply:
The comment 'toggling two bits = ECC failure' is already there. The change was needed because when you poison, you want to toggle parity bits only, so that uncorrected is the same as corrected.

Reply:
The systematic SEC change moved the parity bits.